cppForSwig/Makefile

```diff
@@ -40,23 +42,33 @@ else
 PYTHON_INCLUDES=$(shell $(PYVER)-config --includes )
 endif
 
-CPPFLAGS += $(ARMORY_CPPFLAGS) -Icryptopp -Imdb -DUSE_CRYPTOPP -D__STDC_LIMIT_MACROS
-LDLIBS += -lpthread -Lmdb
-SWIG_OPTS    += -c++ -python -threads
-
-SWIG_INC     += 
-
-CXXCPP += $(CPPFLAGS) $(PYTHON_INCLUDES) -std=c++11
+# "-Wl,--no-as-needed" used for Ubuntu 12.04 due to rt library linking issues.
+# http://stackoverflow.com/questions/17150075/undefined-reference-to-clock-gettime-although-lrt-is-given
+# has more info. Require only for static linking. When 12.04 goes, remove this
+# unless it's discovered that OS X needs it.
+ifdef STATIC_LINK
+LDLIBS += -Wl,--no-as-needed
```

Comment:
Hm, I'm getting a bunch of output like the following after adding -Wl,--no-undefined:

```
CppBlockUtils_wrap.cxx:(.text._ZN4swig19traits_asptr_stdseqISt3setI10BinaryDataSt4lessIS2_ESaIS2_EES2_E5asptrEP7_objectPPS6_[_ZN4swig19traits_asptr_stdseqISt3setI10BinaryDataSt4lessIS2_ESaIS2_EES2_E5asptrEP7_objectPPS6_]+0x3f3): undefined reference to `PyErr_SetString'
```

So it appears to be swig related. I have to go now, so I'll look at it more later.